.github/workflows/ci.yml

```diff
@@ -0,0 +1,117 @@
+name: Run CI
+
+# Run this workflow every time a new commit pushed to your repository
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '*'
+  pull_request:
+  workflow_dispatch:
+
+env:
+  IMAGE_NAME: maykinmedia/record-management-app
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    name: Unit tests
+
+    services:
+      postgres:
+        image: postgres:10
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+        ports:
+          - 5432:5432
+        # needed because the postgres container does not provide a healthcheck
+        options: --health-cmd pg_isready --health-interval 10s --health-timeout 5s --health-retries 5
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.8'
+      - name: Install dependencies
+        run: pip install -r requirements/ci.txt codecov
+      - name: Build frontend
+        run: |
+          npm ci
+          npm run build
+
+      - name: Run tests
+        run: |
+          python src/manage.py collectstatic --noinput --link
+          coverage run src/manage.py test src
+        env:
+          DJANGO_SETTINGS_MODULE: rma.conf.ci
+          SECRET_KEY: dummy
+          DB_USER: postgres
+          DB_PASSWORD: ''
+
+  docker:
+    runs-on: ubuntu-latest
+    name: Docker image build
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Determine tag/commit hash
+        id: vars
+        run: |
+          # Strip git ref prefix from version
+          VERSION=$(echo "${{ github.ref }}" | sed -e 's,.*/\(.*\),\1,')
+          # Strip "v" prefix from tag name (if present at all)
+          [[ "${{ github.ref }}" == "refs/tags/"* ]] && VERSION=$(echo $VERSION | sed -e 's/^v//')
+          # Use Docker `latest` tag convention
+          [ "$VERSION" == "master" ] && VERSION=latest
+          echo ::set-output name=tag::${VERSION}
+          echo ::set-output name=git_hash::${GITHUB_SHA}
+      - name: Build the Docker image
+        run: |
+          docker build \
+            --tag $IMAGE_NAME:${{ steps.vars.outputs.tag }} \
+            --build-arg COMMIT_HASH=${{ steps.vars.outputs.git_hash }} \
+            --build-arg RELEASE=${{ steps.vars.outputs.tag }} \
+            .
+      - run: docker image save -o image.tar $IMAGE_NAME:${{ steps.vars.outputs.tag }}
+      - name: Store image artifact
+        uses: actions/upload-artifact@v2
+        with:
+          name: docker-image
+          path: image.tar
+          retention-days: 1
+
+  publish:
+    needs:
+      - tests
+      - docker
+    name: Push docker image
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'   # Excludes PRs/forks
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download built image
+        uses: actions/download-artifact@v2
+        with:
+          name: docker-image
+
+      - name: Determine tag/commit hash
```

Comment:
I think if you make `docker` and `publish` a one job, this duplicated step can be removed